src/main/java/frc/robot/classes/spikemotor/SpikeMotor.java

```diff
@@ -0,0 +1,252 @@
+    *
+    * @param deviceNumber    an int giving the device number
+    * @return                nothing
+    * @see                   #init(int deviceNumber)
+    */
+    abstract protected void initImpl(int deviceNumber);
+
+    /**
+    * The implementation of setSpeed.
+    * @param speed    a double indicating the speed
+    * @return         nothing
+    * @see            #setSpeed(double speed)
+    */
+    abstract protected void setSpeedImpl(double speed);
+
+    /**
+    * The implementation of getSpeed.
+    *
+    * @return    a double indicating the speed
+    * @see       #getSpeed()
+    */
+    abstract protected double getSpeedImpl();
+
+    /**
+    * The implementation of setPosition.
+    *
+    * @param position   a double indicating the position to set
+    * @return           nothing
+    * @see              #setPosition()
+    */
+    abstract protected void setPositionImpl(double position);
+    
+    /**
+    * The implementation of getPosition.
+    *
+    * @return    a double indicating the position
+    * @see       #getPosition()
+    */
+    abstract protected double getPositionImpl();
+
+    /**
+    * The implementation of moveTo.
+    *
+    * @param position   a double indicating the position to move to
+    * @return           nothing
+    * @see              #moveTo()
+    */
+    abstract protected void moveToImpl(double position);
+
+    /**
+    * Gets the wheel diameter in feet.
+    *
+    * @return   the wheel diameter in feet
+    */
+    abstract protected double getWheelDiameter();
+
+    /**
+    * Accelerates the mover towards the given speed.
+    * @param speed    a double indicating the speed to accelerate to
+    * @param maxStep  a double indicating the maximum speed change
+    * @return         nothing
+    * @see            #accelTowardImpl()
+    */
+    public final void accelToward(double speed, double maxStep) {
+        if (!isInitialized)
+            throw new IllegalStateException(
+                "Motor must be initialized first"
+            );
+        accelTowardImpl(speed, maxStep);
+    }
+
+    /**
+    * The implementation of accelToward.
+    * @param speed    a double indicating the speed to accelerate to
+    * @param maxStep  a double indicating the maximum speed change
+    * @return         nothing
+    * @see            #accelToward()
+    */
+    protected void accelTowardImpl(double speed, double maxStep) {
+        double currentSpeed = getSpeed();
+        double newSpeed = speed;
+        if (speed > currentSpeed) {
+            newSpeed = Math.min(currentSpeed + maxStep, speed);
+        } else {
+            newSpeed = Math.max(currentSpeed - maxStep, speed);
+        }
+        setSpeed(newSpeed);
+    }
+
+    /**
+    * Move to a given angle in radians.
+    * @param angle  a double indicating the angle in radians
+    * @return       nothing
+    */
+    public final void moveToAngle(double angle) {
+        moveTo(Math.PI * getWheelDiameter() * angle);
```

Comment:
MoveToAngle is something that we may want to control in the subsystem and not the motor. The motor class does not know the application that it is used in, it only knows itself, the motor. Think about the shoulder mechanism that will have a gearbox and a pulley. Things are a bit different then. Now picture this is the extendo-arm. An even different story. 

The subsystem knows that there is a motor, gearbox, and pulley in play so it should be responsible for putting the pieces together

Reply:
This should just be a `MoveToPosition` call which sets the motor to an encoder value and maybe the direction

Reply:
The reason that this exists is so that we don't need to worry about the conversion factor. `moveTo` is that call that you're thinking of.

Reply:
The conversion factor will not be the same. There will be motors that do not have a wheel diameter

Reply:
Then why would we be calling this on them?